cirq-superstaq/cirq_superstaq/job.py

```diff
@@ -319,6 +320,44 @@ def input_circuits(
         """
         return self._get_circuits("input_circuit", index=index)
 
+    def pulse_gate_circuits(self, index: Optional[int] = None) -> Any:
+        """Gets the pulse gate circuit returned by this job.
+
+        Returns:
+            The `qiskit.QuantumCircuit` pulse gate circuit.
+
+        Raises:
+            ValueError: If job was not run on an IBM pulse device.
+        """
+        job_ids = self._job_id.split(",")
+
+        if not all(
+            job_id in self._job and self._job[job_id].get("pulse_gate_circuits") is not None
+            for job_id in job_ids
+        ):
+            self._refresh_job()
+
+        if index is None:
+            if all(self._job[job_id].get("pulse_gate_circuits") for job_id in job_ids):
+                serialized_circuits = [
+                    self._job[job_id]["pulse_gate_circuits"] for job_id in job_ids
+                ]
+                deserialized_circuits = []
+                for serialized_circuit in serialized_circuits:
+                    deserialized_circuit = css.serialization.deserialize_qiskit_circuits(
+                        serialized_circuit, False
+                    )
+                    if deserialized_circuit is not None:
+                        deserialized_circuits.append(deserialized_circuit[0])
```

Comment:
if one circuit fails to deserialize, they probably all will so imo it's better to bail out instead of printing the same warning over and over (and otherwise they can always uses `index` to try to deserialize one circuit at a time, which is probably more useful than getting an unspecified subset of the circuits)
```suggestion
                    if deserialized_circuit is None:
                        raise ValueError("Some circuits could not be deserialized.")
                    deserialized_circuits.append(deserialized_circuit[0])
```